tests/aws/aio/aiomoto_fixtures.py

```diff
@@ -0,0 +1,184 @@
+"""
+AWS asyncio test fixtures
+"""
+
+import aiobotocore.client
+import aiobotocore.config
+import pytest
+
+from tests.aws.aio.aiomoto_services import MotoService
+from tests.aws.utils import AWS_ACCESS_KEY_ID
+from tests.aws.utils import AWS_SECRET_ACCESS_KEY
+
+
+#
+# Asyncio AWS Services
+#
+
+
+@pytest.fixture
+async def aio_aws_batch_server():
+    async with MotoService("batch") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_cloudformation_server():
+    async with MotoService("cloudformation") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_ec2_server():
+    async with MotoService("ec2") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_ecs_server():
+    async with MotoService("ecs") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_iam_server():
+    async with MotoService("iam") as svc:
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_dynamodb2_server():
+    async with MotoService("dynamodb2") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_logs_server():
+    # cloud watch logs
+    async with MotoService("logs") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_s3_server():
+    async with MotoService("s3") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_sns_server():
+    async with MotoService("sns") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+@pytest.fixture
+async def aio_aws_sqs_server():
+    async with MotoService("sqs") as svc:
+        svc.reset()
+        yield svc.endpoint_url
+
+
+#
+# Asyncio AWS Clients
+#
+
+
+@pytest.fixture
+def aio_aws_session(aws_credentials, aws_region, event_loop):
+    # pytest-asyncio provides and manages the `event_loop`
+
+    session = aiobotocore.get_session(loop=event_loop)
+    session.user_agent_name = "aiomoto"
+
+    assert session.get_default_client_config() is None
+    aioconfig = aiobotocore.config.AioConfig(
+        max_pool_connections=1, region_name=aws_region
+    )
+
+    # Note: tried to use proxies for the aiobotocore.endpoint, to replace
+    #      'https://batch.us-west-2.amazonaws.com/v1/describejobqueues', but
+    #      the moto.server does not behave as a proxy server.  Leaving this
+    #      here for the record to avoid trying to do it again sometime later.
+    # proxies = {
+    #     'http': os.getenv("HTTP_PROXY", "http://127.0.0.1:5000/moto-api/"),
+    #     'https': os.getenv("HTTPS_PROXY", "http://127.0.0.1:5000/moto-api/"),
+    # }
+    # assert aioconfig.proxies is None
+    # aioconfig.proxies = proxies
```

Comment:
at work I did something like this, may help:

```python
def _wrapt_boto_create_client(wrapped, instance, args, kwargs):
    def unwrap_args(service_name, region_name=None, api_version=None,
                    use_ssl=True, verify=None, endpoint_url=None,
                    aws_access_key_id=None, aws_secret_access_key=None,
                    aws_session_token=None, config=None):

        if endpoint_url is None:
            endpoint_url = os.environ.get(f'{service_name}_mock_endpoint_url')

        # https://github.com/spulec/moto/issues/2058
        aws_access_key_id = "foobar_key"
        aws_secret_access_key = "foobar_secret"

        return wrapped(service_name, region_name, api_version, use_ssl, verify,
                       endpoint_url, aws_access_key_id, aws_secret_access_key,
                       aws_session_token, config)

    return unwrap_args(*args, **kwargs)

def patch_aioboto():
    """
    Will patch aiobotocore to set endpoint_url to: {SERVICE_NAME}_endpoint_url if
    available
    """
    if not isinstance(aiobotocore.session.AioSession.create_client, wrapt.ObjectProxy):
        wrapt.wrap_function_wrapper(
            'aiobotocore.session',
            'AioSession.create_client',
            _wrapt_boto_create_client
        )
```

but ya even better would be with proxies.  It would involve us writing a proxy server that would re-direct to the appropriate endpoint based on the host in the request

Reply:
also had this:

```python
# https://github.com/spulec/moto/issues/2058
for key in {'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY'}:
    if key in os.environ:
        del os.environ[key]

def patch_aioboto():
    """
    Will patch aiobotocore to set endpoint_url to: {SERVICE_NAME}_endpoint_url if
    available
    """
    if not isinstance(aiobotocore.session.AioSession.create_client, wrapt.ObjectProxy):
        wrapt.wrap_function_wrapper(
            'aiobotocore.session',
            'AioSession.create_client',
            _wrapt_boto_create_client
        )
```

Reply:
Proxy would work, but you'd need to patch use_ssl=False right, otherwise it'll try and tunnel through the proxy using CONNECT? And at that point if your patching the ssl flag you might as well patch the endpoint url?

Reply:
Yes, it was trying to tunnel through the proxy with CONNECT.